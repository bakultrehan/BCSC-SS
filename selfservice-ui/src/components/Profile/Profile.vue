/** * Complete Profile of app */

<template>
  <v-card class="mx-auto card-width">
    <v-toolbar flat class="bc-subtitle" dark>
      <v-toolbar-title>{{ $t('profile.pagetitle') }}</v-toolbar-title>
      <div class="flex-grow-1"></div>
    </v-toolbar>
    <v-divider></v-divider>
    <v-form ref="form" v-model="form">
      <v-container>
        <v-row class="ma-5">
          <v-col cols="12" md="12">
            <v-card-subtitle
              class="text-left padding-0 bc-padding-0"
              v-html="$t('profile.titlePageInfo')"
            ></v-card-subtitle>

            <v-card-title class="text-left bc-padding-0">
              {{ userProfile.firstName }}
              {{ userProfile.lastName }}</v-card-title
            >
            <Input
              v-model="email"
              :label="$t('profile.labelEmail')"
              type="text"
              :rules="[
                rules.required,
                rules.email,
                rules.length(2),
                rules.maxLength(250)
              ]"
              v-if="filedsToShow.email"
            />
            <Input
              v-model="phone"
              :label="$t('profile.labelPhone')"
              type="text"
              :rules="[rules.required, rules.length(9), rules.maxLength(15)]"
              v-if="filedsToShow.phone"
            />
          </v-col>
          <v-col cols="12">
            <v-card flat>
              <v-divider></v-divider>
              <v-card-actions>
                <v-spacer></v-spacer>
                <Button
                  :disabled="!form"
                  class="white--text"
                  color="indigo accent-4"
                  depressed
                  @click="completeProfile"
                  @keyup.enter="completeProfile"
                  >{{ $t('profile.btnContinue') }}</Button
                >
              </v-card-actions>
            </v-card>
          </v-col>
        </v-row>
      </v-container>
    </v-form>
  </v-card>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import Input from '@/Atomic/Input/Input.vue';
import Button from '@/Atomic/Button/Button.vue';
import validationRules from '@/config/validationRules';

const KeyCloakModule = namespace('KeyCloakModule');

@Component({ components: { Input, Button } })
export default class Dashboard extends Vue {
  @KeyCloakModule.Getter('filedsToShow')
  private filedsToShow!: any;
  @KeyCloakModule.Action('updateProfile')
  private updateProfile!: any;
  @KeyCloakModule.Getter('userProfile') private userProfile!: [];
  @KeyCloakModule.Action('errorStatus')
  private errorStatus!: any;

  private form: boolean = false;

  /* istanbul ignore next */
  private rules = validationRules;
  private email: string = '';
  private phone: string = '';

  private completeProfile() {
    const profile = { email: this.email, phone: this.phone };
    this.updateProfile(profile);
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.card-width {
  max-width: 80%;
  @include lg {
    max-width: 50%;
  }
}
</style>
