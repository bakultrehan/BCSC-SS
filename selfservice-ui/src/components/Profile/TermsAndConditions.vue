/** * Create or Update Profile of app */

<template>

    <v-container fluid="" >
      <v-row no-gutters="" class="terms">
        <v-col cols="12" md="12">
              <div class="mx-4">
                <p>
                  The BC Services Card Authentication Service (the “Service”) allows you, the entity accepting
                  these Terms of Use (“you”, “You”, “your” or “Your”) to access and use the identity and
                  authentication services of the Provincial Identity Information Management Program (“IDIM”) to
                  deliver your service ( “Your Solution”) to end users. The Service includes: (a) a test environment
                  where you can ensure that Your Solution will integrate and function correctly with the Service;
                  and (b) a production environment to which you may be given access to deploy Your Solution
                  following an assessment by IDIM of Your Solution and the execution of an Information Sharing
                  Agreement to address any “personal information” (as that term is defined in the Freedom of
                  Information and Protection of Privacy Act (B.C.)) that may be exchanged between the Service
                  and Your Solution.
                </p>
                <p>
                  These Terms of Use are an agreement between you and Her Majesty the Queen in Right of the
                  Province of British Columbia, represented by the Minister of Citizens’ Services, IDIM (the
                  “Province”), and govern your use of the Service. By indicating that you agree to these Terms of
                  Use, and in consideration of the use of the Service, you agree to the following:
                </p>
                <h3>1. Authority and Ability to Accept Terms of Use</h3>
                <p>
                  The individual accepting these Terms of Use on your behalf represents and warrants that: (a) s/he
                  is at least 19 years of age; and (b) s/he has all necessary authority to accept these Terms of Use
                  on your behalf.
                </p>
                <h3>2. Responsibility for Users</h3>
                <p>
                  You acknowledge and agree that you are responsible for ensuring that any of your employees,
                  agents and representatives (collectively, “Users”) who access or use the Service comply with
                  these Terms of Use.
                </p>
                <h3>3. Ownership</h3>
                <p>
                  The Service, the website through which the Service is accessed, and all other information and
                  works made available, displayed or transmitted in connection with the Service, including without
                  limitation data, text, audio, video, trademarks, trade names, logos, domain names, images,
                  graphics, graphical user interface elements and designs, in any form or medium whatsoever
                  (collectively, the “Content”), are owned by the Province or its licensors and are protected by
                  copyright, patent, trademark and other laws protecting intellectual property rights.
                </p>
                <h3>4. License</h3>
                <p>
                  Subject to your compliance with these Terms of Use, the Province grants you a non-exclusive,
                  revocable, limited license to access and use the Service and the Content solely as necessary to
                  test and, if applicable, to deploy Your Solution in accordance with these Terms of Use. You must
                  not copy, distribute, republish, transmit or otherwise exploit any part of the Service or the Content,
                  and the Province expressly reserves all intellectual property rights in and to the Service and the
                  Content. If your use of the Service is terminated under section 5, your license is automatically
                  terminated
                </p>
                <h3>5. Suspension or Termination</h3>
                <p>
                  The Province may at any time, in its sole discretion, suspend or terminate your use of the Service,
                  in whole or in part: (a) if you fail to comply with any provision of these Terms of Use; (b) as a
                  security measure; or (c) for administrative purposes.
                </p>
                <h3>6. Changes to Service and/or Terms of Use</h3>
                <p>
                  The Province may at any time, in its sole discretion and without direct notice to you: (a) discontinue
                  the Service; or (b) make changes to the Service and/or these Terms of Use. By continuing to use
                  the Service, you will be conclusively deemed to have accepted any changes to these Terms of
                  Use.
                </p>

                <h3>7. Acceptable Use</h3>
                <p>
                  You must not take any action in connection with your use of the Service that would jeopardize the
                  security, integrity and/or availability of the Service, including without limitation:
                    <ul>
                      <li>(a) using the Service for any unlawful or inappropriate purpose;
                      </li>
                      <li>(b) decompiling, disassembling, reverse engineering or otherwise copying any
                          software associated with the Service;
                      </li>
                      <li>(c) tampering with any portion of the Service;
                      </li>
                      <li>(d) using the Service to transmit any virus or other harmful or destructive computer
                          code, files or programs or to conduct hacking and/or intrusion activities;
                      </li>
                      <li>(e) attempting to circumvent or subvert any security measure associated with the
                          Service;
                      </li>
                      <li>(f) taking any action that might reasonably be construed as likely to adversely affect
                          other users of the Service; or
                      </li>
                      <li>(g) removing or altering any proprietary symbol or notice, including any copyright
                          notice, trademark or logo, displayed in connection with the Service.
                      </li>
                    </ul>

                </p>

                <h3>8. Privacy</h3>
                <p>
                  When you use the Service, “contact information” (as that term is defined in the Freedom of
                  Information and Protection of Privacy Act (B.C.)) is collected from the individual accepting these
                  Terms of Use on your behalf. If you wish to be granted access to the production environment,
                  you acknowledge that: (a) such access is at the sole discretion of the Province; (b) you will be
                  required to undergo an assessment of Your Solution; and (b) you must enter into an Information
                  Sharing Agreement to ensure that any exchange of information between the Service and Your
                  Solution is compliant with applicable legislation, including without limitation the Freedom of
                  Information and Protection of Privacy Act (B.C.).
                </p>
                <h3>9. Service Provided “As Is”</h3>
                <p>
                  The Service is provided to you “as is”, and the Province disclaims all representations, warranties,
                  conditions, obligations and liabilities of any kind, whether express or implied, in relation to the
                  Service, including without limitation implied warranties with respect to merchantability, satisfactory
                  quality, fitness for a particular purpose and non-infringement. Without limiting the general nature
                  of the previous sentence, the Province does not represent or warrant that:
                 <ul>
                    <li>(a) the Service will be available;</li>
                    <li>((b) your use of the Service will be timely, uninterrupted or error free;
                    </li>
                    <li>(c) any errors in the Service will be corrected; or
                    </li>
                    <li>(d) the Service will meet your expectations and requirements.
                    </li>
                  </ul>
                </p>
                <h3>10. No Liability</h3>
                <p>
                  To the maximum extent permitted by applicable law, under no circumstances will the Province be
                  liable to any person or entity for any direct, indirect, special, incidental, consequential or other
                  loss, claim, injury or damage, whether foreseeable or unforeseeable (including without limitation
                  claims for damages for loss of profits or business opportunities, use or misuse of, or inability to
                  use, the Service, interruptions, deletion or corruption of files, loss of programs or information,
                  errors, defects or delays) arising out of or in any way connected with your use of the Service and
                  whether based on contract, tort, strict liability or any other legal theory. The previous sentence
                  will apply even if the Province has been specifically advised of the possibility of any such loss,
                  claim, injury or damage
                </p>
                <h3>11. Indemnity</h3>
                <p>
                  You agree to indemnify, defend and hold harmless the Province and the Province’s employees
                  and agents from and against all claims, demands, obligations, losses, liabilities, costs or debts,
                  and expenses (including but not limited to reasonable legal fees) arising from: (a) your, and your
                  Users’, use of the Service; or (b) your, or any User’s, violation of any provision of these Terms of
                  Use.
                </p>
                <h3>12. General</h3>
                <p>
                  These Terms of Use are the entire agreement between you and the Province with respect to the
                  subject matter of these Terms of Use. The headings in these Terms of Use are inserted for
                  convenience only and will not be used in interpreting or construing any provision of these Terms
                  of Use. If any provision of these Terms of Use is invalid, illegal or unenforceable, that provision
                  will be severed from these Terms of Use and all other provisions will remain in full force and effect.
                  These Terms of Use will be governed by and construed in accordance with the laws of the
                  province of British Columbia and the applicable laws of Canada. By using the Service, you
                  consent to the exclusive jurisdiction and venue of the courts of the province of British Columbia,
                  sitting in Victoria, for the hearing of any dispute arising from or related to these Terms of Use and
                  their subject matter.
                </p>
              </div>
        
        </v-col>
      </v-row>
    </v-container>

</template>

<script lang="ts">
import { Component, Vue, Prop, Watch } from 'vue-property-decorator';

@Component
export default class TermsAndConditions extends Vue {}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.card-width {
  max-width: 80%;
  @include lg {
    max-width: 50%;
  }
}
.terms p{
font-size: 16px;
}

</style>
