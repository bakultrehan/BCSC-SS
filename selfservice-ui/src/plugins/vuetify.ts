import Vue from 'vue';
import Vuetify from 'vuetify/lib';

Vue.use(Vuetify);

export default new Vuetify({
  icons: {
    iconfont: 'mdi',
  },
  theme: {
    themes: {
      light: {
        error: '#D8292F',
        success: '#2E8540',
        warning: '#FFC107',
      },
    },
    options: {
      cspNonce: 'cLv1G2y8KtMdR',
    },
  },
});
