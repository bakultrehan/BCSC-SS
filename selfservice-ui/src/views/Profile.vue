<template>
  <div class="home">
    <Profile />
  </div>
</template>

<script lang="ts">
// @ is an alias to /src
import { Component, Vue, Prop } from 'vue-property-decorator';
import Profile from '@/components/Profile/Profile.vue';

@Component({
  components: {
    Profile
  }
})
export default class ProfileHome extends Vue {}
</script>
