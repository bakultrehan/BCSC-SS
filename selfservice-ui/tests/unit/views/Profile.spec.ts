import { shallowMount, createLocalVue } from '@vue/test-utils';
import Profile from '@/views/Profile.vue';
import vuetify from 'vuetify';

describe('Profile.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(vuetify);

    wrapper = shallowMount(Profile, {
      localVue
    });
  });

  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
