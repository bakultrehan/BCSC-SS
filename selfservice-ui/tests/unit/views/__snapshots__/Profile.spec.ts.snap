// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile.vue renders props when passed 1`] = `
<div
  class="home"
>
  <profile-stub />
</div>
`;
